app/services/claims/fee_calculator/price.rb

```diff
@@ -28,7 +28,7 @@ def unit
       def modifiers
         @modifiers ||= price.modifiers.each_with_object([]) do |modifier, arr|
           modifier = ModifierDecorator.new(modifier)
-          arr.append(modifier) if unit_modifiers.any? { |unit_modifier| modifier == unit_modifier }
+          arr.append(modifier) if unit_modifiers.any?(modifier)
```

Comment:
This line is causing the test failures. The old version, which is flagged by `Performance/RedundantEqualityComparisonBlock`, is:

```ruby
arr.append(modifier) if unit_modifiers.any? { |unit_modifier| modifier == unit_modifier }
```

and `modifier == unit_modifier` is not the same as `unit_modifier == modifier` because `modifier` and `unit_modifier` are not the same type and the definitions of `==` differ.

I'll have a think about what is the best thing to do here.

Reply:
@mabely if you had time and wanted to take a look...

Reply:
We could alias `===` to `==` in the [`modifierDecorator`](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/blob/fa36e799d25899ae001e8610e1146d6c8f59327d/app/services/claims/fee_calculator/modifier_decorator.rb#L16). It doesn't look like `===` is being used for elsewhere, and the tests and Rubocop pass. 

I assume `unit_modifiers` is an array of `modifier` objects, but the more I look at the [Rubocop documentation](https://docs.rubocop.org/rubocop-performance/cops_performance.html#performanceredundantequalitycomparisonblock) and dig into [Rubocop code](https://github.com/rubocop/rubocop-performance/blob/master/lib/rubocop/cop/performance/redundant_equality_comparison_block.rb#L25), the less I understand it seems...

> This cop checks for uses Enumerable#all?, Enumerable#any?, Enumerable#one?, and Enumerable#none? are compared with === or similar methods in block.

> By default, Object#=== behaves the same as Object#==, but this behavior is appropriately overridden in subclass. For example, Range#=== returns true when argument is within the range. Therefore, It is marked as unsafe by default because === and == do not always behave the same.

Reply:
I have reverted this change and added an exception to `.rubocop_todo.yml`